server/pulp/server/webservices/views/serializers/__init__.py

```diff
@@ -0,0 +1,163 @@
+        :type instance: object
+        :return: serialized form of the object
+        :rtype: dict
+        """
+        raise NotImplementedError
+
+    def get_href(self, instance):
+        """
+        Get the href for the instance being serialized. If none is returned then
+        no href will be added to the serialized object
+
+        :param instance: Instance of the object to be serialized
+        :type instance: object
+        :return: The href for the object being serialized
+        :rtype: str
+        """
+        return None
+
+    def _to_representation(self, instance):
+        """
+        Internal method to turn an instance into it's serialized representation
+
+        :param instance: the object to be serialized
+        :type instance: object
+        :return: The instance object represented as a dictionary ready for jsonification
+        :rtype: dict
+        """
+        representation = self.to_representation(instance)
+        if self._exclude_fields:
+            for blacklist_field in self._exclude_fields:
+                field_accessor = blacklist_field.split('__')
+                self._remove_excluded(field_accessor, representation)
+        if self._mask_fields:
+            for pwd_field in self._mask_fields:
+                field_accessor = pwd_field.split('__')
+                self._mask_field(field_accessor, representation)
+
+        href = self.get_href(instance)
+        if href:
+            representation['_href'] = href
+
+        return representation
+
+    def _remove_excluded(self, accessor, representation):
+        """
+        Internal method to remove excluded fields from the dictionary form of an instance.
+
+        This takes a single accessor and
+
+        :param accessor: The accessor to the particular field to be removed
+        :type accessor: list of str
+        :param representation: The representation of the instance being serialized
+        :type representation: dict
+        """
+        root_key = accessor[0]
+        if root_key in representation:
+            if len(accessor) > 1:
+                self._remove_excluded(accessor[1:], representation[root_key])
+            else:
+                representation.pop(root_key, None)
+
+    def _mask_field(self, accessor, representation):
+        """
+        Internal method to replace password values with a fixed number of asterisks
+        during serialization.
+
+        :param accessor: The accessor to the particular field to be removed
+        :type accessor: list of str
+        :param representation: The representation of the instance being serialized
+        :type representation: dict
+        """
+        root_key = accessor[0]
+        if root_key in representation:
+            if len(accessor) > 1:
+                self._mask_field(accessor[1:], representation[root_key])
+            else:
+                representation[root_key] = '*****'
+
+    @property
+    def data(self):
```

Comment:
So, the reason this uses serializer.data is so that we could think about adding a create() or update() method similar to http://www.django-rest-framework.org/api-guide/serializers/ to support taking a raw data dictionary from the request and turning it into a mongoengine model for use in other calls.